File: java/src/main/java/accounting/patterns/EventType.java
package accounting.patterns;

public class EventType {
    public static EventType USAGE = eventType("usage");
    public static EventType SERVICE = eventType("service fee");
    private final String name;

    public EventType(String name) {
        this.name = name;
    }

    public static EventType eventType(String name) {
        return new EventType(name);
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return "EventType{" +
                "name='" + name + '\'' +
                '}';
    }
}

File: src/main/typescript/accounting-pattern/EventType.ts
/** The kind of event a customer posts against a service agreement. */
export class EventType {
    static readonly USAGE = new EventType("usage");
    static readonly SERVICE = new EventType("service fee");

    private readonly name: string;

    constructor(name: string) {
        this.name = name;
    }

    static eventType(name: string): EventType {
        return new EventType(name);
    }

    getName(): string {
        return this.name;
    }

    toString(): string {
        return `EventType{name='${this.name}'}`;
    }
}
